corehq/motech/migrations/0017_connectionsettings_use_aes_cbc_encryption.py

```diff
@@ -70,12 +71,59 @@ def reencrypted_last_token_with_cbc(connection):
         return ''
 
 
+def revert_api_settings(apps, schema_editor):
+    ConnectionSettings = apps.get_model("motech", "ConnectionSettings")
+
+    connect_settings_to_revert = ConnectionSettings.objects.exclude(
+        password__startswith=f'${ALGO_AES}$',
+        client_secret__startswith=f'${ALGO_AES}$',
+        last_token_aes__startswith=f'${ALGO_AES}$',
```

Comment:
Same issue when reencrypt from ECB to CBC, the mixed state is not handled.

Reply:
I think the `_revert` functions correctly handle the mixed state here. There are five scenarios
1. Value is already migrated and contain the `aes-cbc` prefix
2. Value is not migrated and is EBC encrypted and contains the `aes` prefix
3. Value is not migrated and is EBC encrypted but does not contain the `aes` prefix
4. Value is not migrated and is not encrypted
5. Value is empty (`''`)

The _revert functions handles each as such:
1. reencrypts the value from cbc to ecb mode
2. returns the original unmigrated value
3. returns the original unmigrated value
4. returns the original unmigrated value
5. returns `''`